### ISSUE TITLE
Missing/moved `wa` library in 0.0.34 breaking keybindings

### ISSUE THREAD

**Issue by destructobeam** [raphamorim/rio#442]
Hi. Keybindings that are working on macOS in 0.0.33 have broken in 0.0.34 (for example `Command+Shift+[` and `]`; binding manually in the config also doesn't work). Looks like this is due to the missing `wa` library, which I see from other issues you are migrating away from for 0.1.0.

Would it be possible to get a 0.0.35 release with that back (looks like nightly has it) in the meantime please?

Thanks!

**Comment by raphamorim** [comment]
Hey there @destructobeam thanks for the issue!

Sorry about the delay have been working on #428 , couldn't reply much issues.

Hmm weird, I can't reproduce it, the command+shift+[ or ] does work for me in the 0.0.34. What's your configuration file?

**Comment by destructobeam** [comment]
Thanks @raphamorim, I don't have any custom config for keybindings at the moment, as they are working by default in 0.0.33. I have tried both the following in 0.0.34 without success however, on macOS 14.3:

```toml
[bindings]
keys = [
  { key = "[", with = "super + shift", action = "SelectPrevTab" },
  { key = "]", with = "super + shift", action = "SelectNextTab" }
]
```
```toml
[bindings]
keys = [
  { key = "{", with = "super", action = "SelectPrevTab" },
  { key = "}", with = "super", action = "SelectNextTab" }
]
```

**Comment by destructobeam** [comment]
Also, just to be thorough, I tried the following in 0.0.34 as well with no luck:
```toml
[bindings]
keys = [
  { key = "[", with = "command + shift", action = "SelectPrevTab" },
  { key = "]", with = "command + shift", action = "SelectNextTab" }
]
```

**Comment by destructobeam** [comment]
I don't mean to nag you or anything, I only just saw that there was a 0.0.35 release. Confirming the above is not working in 0.0.35 either.

**Comment by MrPandir** [comment]
I've been using this combination since version [v0.0.38](https://github.com/raphamorim/rio/releases/tag/v0.0.38) and it works fine in all versions. Your problem is that you use `+` for the combination, but you need to use `|`, as described in the [documentation](https://raphamorim.io/rio/docs/next/key-bindings#with).
Here is the working solution I use:
```toml

[bindings]
keys = [
    { key = "{", with = "super | shift", action = "SelectPrevTab" },
    { key = "}", with = "super | shift", action = "SelectNextTab" },
]
```

@raphamorim Perhaps rio should report errors in the config when it finds unknown symbols or words that are not reserved and refer to the documentation.

**Comment by raphamorim** [comment]
Regarding the error message i agree, it’s a good idea report invalid key bindings 👍

**Comment by MrPandir** [comment]
> Is it working on 0.1.6?

Yes it works in 0.1.6

I wonder why this doesn't work by default and requires adding it to the config?
Judging by the code, this should work by default. 🤔
https://github.com/raphamorim/rio/blob/6f6011ee2aec6be36fcdf0e203608d327a90386e/frontends/rioterm/src/bindings/bindings_wa/mod.rs#L944-L945

**Comment by raphamorim** [comment]
Yea, that's the weirdest thing. Before it was a bug that was computing only if was `{` or `}` unfortunately. This bug was introduce somewhere 0.0.33~0.0.34 and stayed in the source code a while. I made a fix for it only in 0.1.4, so technically you shouldn't need anymore `{` or `}` bindings. It is working for me without any extra stuff by default.

**Comment by MrPandir** [comment]
> Yea, that's the weirdest thing. Before it was a bug that was computing only if was `{` or `}` unfortunately. This bug was introduce somewhere 0.0.33~0.0.34 and stayed in the source code a while. I made a fix for it only in 0.1.4, so technically you shouldn't need anymore `{` or `}` bindings. It is working for me without any extra stuff by default.

Oh, yes, in `0.1.6` it works by default and does not require changing the config. Good job. ❤️

**Comment by destructobeam** [comment]
@raphamorim Thanks, this is back to working without any binding config needed. 👍
